be/src/exec/pipeline/pipeline_driver.cpp

```diff
@@ -302,23 +302,13 @@ void PipelineDriver::finalize(RuntimeState* runtime_state, DriverState state) {
     COUNTER_UPDATE(_schedule_timer, _total_timer->value() - _active_timer->value() - _pending_timer->value());
     _update_overhead_timer();
 
-    // last root driver cancel the all drivers' execution and notify FE the
-    // fragment's completion but do not unregister the FragmentContext because
-    // some non-root drivers maybe has pending io io tasks hold the reference to
-    // object owned by FragmentContext.
-    if (is_root()) {
-        if (_fragment_ctx->count_down_root_drivers()) {
-            _fragment_ctx->finish();
-            auto status = _fragment_ctx->final_status();
-            _fragment_ctx->runtime_state()->exec_env()->driver_executor()->report_exec_state(_fragment_ctx, status,
-                                                                                             true);
-        }
-    }
```

Comment:
If `is_report_success` is false, could we use the original logic, which reports as soon as root finished?

Reply:
「the elapsed time between root finished and all the drivers finished may be long」Is this conclusion based on tests? In my opinion, when root driver is finished, the other non-root drivers are doing the finishing jobs, theoretically, it won't take too long. 

Considering the extreme situation you point out, exec state will be sent as soon as root driver finished, while other non-root drivers are busy doing finishing jobs. The early return cannot actually release the resources it occupied (by non-root drivers), which means the next bunch of queries may not benefit from the early return in concurrency situations.

 I'll add some tests to performance decline.

Reply:
Test reporter: https://starrocks.feishu.cn/docs/doccnAPF6B71Nct1hql7tu3BtYb#